libs/sdk-ui-dashboard/src/model/store/catalog/catalogSelectors.ts

```diff
@@ -362,6 +361,11 @@ function buildDateHierarchy(
     if (!isEmpty(dateAttributesInHierarchy)) {
         return {
             type: "dateAttributeHierarchy",
+            // create ref of adhoc date hierarchy by combining date dataset id and date template id
```

Comment:
Maybe add there more notes that this id must not be persisted anywhere and is just for the sake of the components that require to have ref.